sm-core-model/src/main/java/com/salesmanager/core/model/catalog/category/Category.java

```diff
@@ -124,12 +115,25 @@ public void setAuditSection(AuditSection auditSection) {
     }
 
 
-    public String getCategoryImage() {
-        return categoryImage;
+    public Set<CategoryImage> getImages() {
+        return categoryImages;
     }
 
-    public void setCategoryImage(String categoryImage) {
-        this.categoryImage = categoryImage;
+    public void setImages(Set<CategoryImage> images) {
+        this.categoryImages = images;
+    }
+
+    public CategoryImage getCategoryImage() {
+        CategoryImage categoryImage = null;
+        if(this.getImages()!=null && this.getImages().size()>0) {
```

Comment:
you don't need to check for null here, you already giving a default value to the set.
also you can do categoryImages.stream().filter(CategoryImage::isDefaultImage()).findFirst() but be careful as it returns optional (which is much better)

Reply:
Yes, you are right we can do that!
But I want to add a general note to be considered while you review the code, I have implemented this feature same as product , with the same way of implementation, so once we decide to refactor the code there will not be any problems,  we will refactor it all.

I will comment for each point you mentioned.